test_unstructured/partition/test_ndjson.py

```diff
@@ -0,0 +1,299 @@
+"""Test-suite for `unstructured.partition.json` module."""
+
+from __future__ import annotations
+
+import os
+import pathlib
+import tempfile
+
+import pytest
+from pytest_mock import MockFixture
+
+from test_unstructured.unit_utils import example_doc_path
+from unstructured.documents.elements import CompositeElement
+from unstructured.file_utils.model import FileType
+from unstructured.partition.email import partition_email
+from unstructured.partition.html import partition_html
+from unstructured.partition.ndjson import partition_ndjson
+from unstructured.partition.text import partition_text
+from unstructured.partition.xml import partition_xml
+from unstructured.staging.base import elements_to_ndjson
+
+DIRECTORY = pathlib.Path(__file__).parent.resolve()
+
+is_in_docker = os.path.exists("/.dockerenv")
+
+test_files = [
+    "fake-text.txt",
+    "fake-html.html",
+    "eml/fake-email.eml",
+]
+
+is_in_docker = os.path.exists("/.dockerenv")
+
+
+def test_it_chunks_elements_when_a_chunking_strategy_is_specified():
+    chunks = partition_ndjson(
+        example_doc_path("spring-weather.html.ndjson"),
+        chunking_strategy="basic",
+        max_characters=1500,
+    )
+
+    assert len(chunks) == 9
+    assert all(isinstance(ch, CompositeElement) for ch in chunks)
```

Comment:
In the general case, chunks can be `CompositeElement`, `Table`, or `TableElement`. Not sure if there's a later case that exercises table behaviors but probably a good idea to have at least one.

Reply:
I copied all these tests verbatim from the json tests but updated it to use the ndjson support.